Migrated js files to tsx

winniebuna opened the pull request.

winniebuna committed:
Migrated js files to tsx

the-jossie commented on components/article/index.tsx:
files without a react component should have `.ts` extension.

while files whose content contains a react component should have `.tsx` extension

please update this for other similar files

winniebuna committed:
Updated file extensions to .ts .tsx

Updated files without a react component to .ts extension.

Updated component files to .tsx extension.

winniebuna committed:
Revert "Updated file extensions to .ts .tsx"

This reverts commit 08011803275a0a9d47c554dc2da242cfcf323529.

winniebuna committed:
Revert "Revert "Updated file extensions to .ts .tsx""

This reverts commit d265f1e842406b759bba9a9ca3d2129617f7d41c.

winniebuna committed:
Update tsconfig.json

winniebuna committed:
Update tsconfig.json

winniebuna committed:
Update tsconfig.json

winniebuna committed:
Update tsconfig.json

winniebuna committed:
Revert "Update tsconfig.json"

This reverts commit acf64852ca60f0f70ab68ff7e997811ed9d03173.

the-jossie committed:
fix: build typescript errors
